testing/SetupTestbed.ps1

```diff
@@ -366,13 +380,63 @@ it is likely that the DNS entry was still successfully added. To
 verify, log on to DC1 and run 'Resolve-DnsName ls1' in PowerShell.
 If it returns NXDOMAIN, you'll need to add it manually."
 Write-Output "The time is $(Get-Date)."
-az vm run-command create `
+
+# Define the PowerShell script with the DomainName variable interpolated
+$scriptContent = @"
+`$scriptBlock = {
+    Add-DnsServerResourceRecordA -Name LS1 -ZoneName $DomainName -AllowUpdateAny -IPv4Address $LsIP -TimeToLive 01:00:00 -AsJob
+}
+`$job = Start-Job -ScriptBlock `$scriptBlock
+`$timeout = 60
+if (Wait-Job -Job `$job -Timeout `$timeout) {
+    Receive-Job -Job `$job
+    Write-Host 'The script completed within the timeout period.'
+} else {
+    Stop-Job -Job `$job
+    Remove-Job -Job `$job
+    Write-Host 'The script timed out after `$timeout seconds.'
+}
+"@
+
+# Convert the script to a Base64-encoded string
+$bytes = [System.Text.Encoding]::Unicode.GetBytes($scriptContent)
+$encodedScript = [Convert]::ToBase64String($bytes)
+
+
+# Run the encoded script on the Azure VM
+Write-Output "`nAdding script to add DNS entry for Linux server. No output expected..."
+$createDnsScriptResponse = az vm run-command invoke `
+    --command-id RunPowerShellScript `
+    --name DC1 `
+    --resource-group $ResourceGroup `
+    --scripts "Set-Content -Path 'C:\AddDnsRecord.ps1' -Value ([System.Text.Encoding]::Unicode.GetString([System.Convert]::FromBase64String('$encodedScript')))"
+
+Show-FormattedOutput -FormattedOutput (Format-AzVmRunCommandOutput -JsonResponse "$createDnsScriptResponse")
+
+
+Write-Output "`nRunning script to add DNS entry for Linux server. It could time out or not. Check output of the next command..."
+$addDnsRecordResponse = az vm run-command invoke `
+    --command-id RunPowerShellScript `
+    --name DC1 `
+    --resource-group $ResourceGroup `
+    --scripts "C:\AddDnsRecord.ps1"
+Show-FormattedOutput -FormattedOutput (Format-AzVmRunCommandOutput -JsonResponse "$addDnsRecordResponse")
+
+Write-Host "Checking if ls1 resolves. This should resolve to ls1.lme.local, not another domain..."

```

Comment:
The DNS call should resolve to an IP address, not a domain name.
```suggestion
Write-Host "Checking if ls1 resolves. This should resolve to $($LsIP)."

```

Reply:
Actually, if the script failed it will show ls1.blablablababadflaflasd.something.net instead of ls1.lme.local. 
Not sure how to word it to indicate that in our output. But we should also let them know it should resolve to that ip. 
It does resolve to that ip even if it is the ls1.blabla one.

Reply:
Interesting! On my test this morning I actually got the ls1.blabla... domain. But I just tested, even if it shows that, everything should still work as expected.

What I tested:
From the Linux server:
![image](https://github.com/cisagov/LME/assets/106177711/8faf1314-fe74-446f-8484-d2ef4ee3a12c)

From one of the clients:
![image](https://github.com/cisagov/LME/assets/106177711/2d963fc9-3bb8-4981-b830-b67463650bbd)

You'll note the weird domain showing under the "Name" column, but we're still able to connect to ls1 despite it.

I wish we understood what the underlying problem here was, but even without that, things should still work as expected.

Reply:
I was running scp on admin.ackbar@ls1.lme.local later in the script and it didn't resolve. I changed it to admin.ackbar@ls1 but I am not happy about that. :)

Reply:
Ah, that's fair. Ok, new proposal then:

Option 1: 
- You change it to just admin.ackbar@ls1 and we have the setup script tell the user to just make sure ls1 resolves to 10.1.0.5.
- Pro: You won't need to manually add the DNS record if we're unable to do it programmatically

Option 2:
- We change the setup script's final check to `Resolve-Dns ls1.lme.local` instead of just `Resolve-Dns ls1`
- Pro: It will cause the check to throw an actual error if the DNS record isn't properly added. Then we could then use the fully-qualified domain name later on, which I believe our instructions actually explicitly say to do in some places.

Reply:
So far this works.